honor CFLAGS, LDFLAGS, and LOCALBASE
use parentheses to not clash with SUBST_CMD
remove -lstdc++ (not needed)

Index: Makefile
--- Makefile.orig
+++ Makefile
@@ -1,30 +1,12 @@
-LBITS := $(shell getconf LONG_BIT)
+CFLAGS = ${CFLAGS} -Wall -I src -I native/include -fPIC -I ../sdk/public -I${LOCALBASE}/include
+#LFLAGS = ${LDFLAGS} -lhl -lsteam_api -lstdc++ -L${LOCALBASE}/lib
+LFLAGS = ${LDFLAGS} -lhl -lsteam_api -L${LOCALBASE}/lib
 
-UNAME := $(shell uname)
-
-CFLAGS = -Wall -O3 -I src -I native/include -fPIC -I ../sdk/public
-
-ifndef ARCH
-	ARCH = $(LBITS)
-endif
-
-LIBARCH=$(ARCH)
-ifeq ($(UNAME),Darwin)
-OS=osx
-# universal lib in osx32 dir
-LIBARCH=32
-else
-OS=linux
-CFLAGS += -std=c++0x
-endif
-
-LFLAGS = -lhl -lsteam_api -lstdc++ -L native/lib/$(OS)$(LIBARCH) -L ../sdk/redistributable_bin/$(OS)$(ARCH)
-
 SRC = native/cloud.o native/common.o native/controller.o native/friends.o native/gameserver.o \
 	native/matchmaking.o native/networking.o native/stats.o native/ugc.o
 
 all: ${SRC}
-	${CC} ${CFLAGS} -shared -o steam.hdll ${SRC} ${LFLAGS}
+	$(CC) $(CFLAGS) -shared -o steam.hdll ${SRC} $(LFLAGS)
 
 install:
 	cp steam.hdll /usr/lib
@@ -33,7 +15,7 @@ install:
 .SUFFIXES : .cpp .o
 
 .cpp.o :
-	${CC} ${CFLAGS} -o $@ -c $<
+	$(CC) $(CFLAGS) -o $@ -c $<
 	
 clean_o:
 	rm -f ${SRC}
